### PR TITLE
feat: sign oracle data

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- We should handle sign oracle data requests

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/WalletConnect/WalletConnectModal.js [hunk 1]
```diff
@@ -43,6 +44,14 @@ export default () => {
             onDismiss={onDismiss}
           />
         );
+      case WalletConnectModalTypes.SIGN_ORACLE_DATA:
+        return (
+          <SignOracleDataModal
+            {...walletConnectModal}
+            onDismiss={onDismiss}
+            baseStyles={baseStyles}
```
Comment (pedroferreira1) [inline src/components/WalletConnect/WalletConnectModal.js:52]: Should we remove it as we removed the others?

File: src/components/WalletConnect/SignOracleDataRequest.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+);
+
+export const SignOracleDataRequest = ({ signOracleData }) => {
+  const { dapp, data } = signOracleData;
+  const { message, address } = data;
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  const onAcceptSignOracleDataRequest = () => {
+    const acceptedNc = { address, message };
```
Comment (pedroferreira1) [inline src/components/WalletConnect/SignOracleDataRequest.js:54]: `acceptedData`?

**Review by pedroferreira1 — APPROVED** [review]

File: src/components/WalletConnect/WalletConnectModal.js (hunk 1 above)
Comment (andreabadesso) [reply src/components/WalletConnect/WalletConnectModal.js]: Done, thanks!

File: src/components/WalletConnect/SignOracleDataRequest.js (hunk 2 above)
Comment (andreabadesso) [reply src/components/WalletConnect/SignOracleDataRequest.js]: Done! Thanks

**Pull request merged**